### PR TITLE
(feat) webrtcvideo: Fully implement initial P2P video

### PR DESCRIPTION
**Pull request by misteryeo**
- Create offer and answer between local and remote peers

### CODE REVIEW

**Comment by misteryeo** [comment]
[Start WebRTC chat module](https://trello.com/c/AEnZZBfq/47-start-webrtc-chat-module)

File: client/components/room/Video.jsx [hunk 1]
```diff
@@ -27,192 +33,285 @@ class Video extends React.Component {
...
-    this.handleVideoScreen = this.handleVideoScreen.bind(this);
-    this.handleAudio = this.handleAudio.bind(this);
+    // this.requestTurn = this.requestTurn.bind(this);
+    this.handleRemoteStreamAdded = this.handleRemoteStreamAdded.bind(this);
+    this.handleRemoteStreamRemoved = this.handleRemoteStreamRemoved.bind(this);
+    this.hangup = this.hangup.bind(this);
+    this.handleRemoteHangup = this.handleRemoteHangup.bind(this);
+    this.stop = this.stop.bind(this);
+    // this.handleVideoScreen = this.handleVideoScreen.bind(this);
+    // this.handleAudio = this.handleAudio.bind(this);
```
Comment (dshelters) [inline client/components/room/Video.jsx]: Just wondering if we are keeping these bindings here for potential later use, or if we can delete the commented out lines here.

File: client/components/room/Video.jsx [hunk 2]
```diff
@@ -27,192 +33,285 @@ class Video extends React.Component {
...
   }
 
   componentDidMount() {
-    socket.emit('Hello', 'Hello');
-    const localVideo = document.getElementById('localVideo');
-    const remoteVideo = document.getElementById('remoteVideo');
+    let localVideo = document.getElementById('localVideo');
+    let remoteVideo = document.getElementById('remoteVideo');
```
Comment (dshelters) [inline client/components/room/Video.jsx]: While not MVP, I think there might be a way to set src on these elements without using getElementById.

File: client/components/room/Video.jsx [hunk 3]
```diff
@@ -27,192 +33,285 @@ class Video extends React.Component {
...
+  //       video: this.state.constraints.video,
+  //       audio: !this.state.constraints.audio
+  //     },
+  //     mute: !this.state.constraints.audio ? 'Mute' : 'Unmute'
+  //   }, () => (this.start()));
+  //   navigator.mediaDevices.getUserMedia(this.state.constraints)
+  //     .then(this.successCallback)
+  //     .catch(this.errorCallback);
+  // }
+
```
Comment (dshelters) [inline client/components/room/Video.jsx]: Same thing as with the function bindings. Wondering if we can remove this commented block before merging.

**Comment by jefQuery** [comment]
🙌
🙌
🙌

File: client/components/room/Video.jsx [hunk 4]
```diff
@@ -2,17 +2,23 @@ import React from 'react';
...
+let turnReady;
+let pc;
+
+const configuration = {
+  'iceServers': [{
+    'urls': 'stun:stun.l.google.com:19302'
+  }]
+};
+
+let sdpConstraints = {
```
Comment (dbrodie122) [inline client/components/room/Video.jsx]: In a later pull, could we get some small comments that give explanation for what these lingo-heavy sections do?

**Pull request merged**